src/main/java-intellij/com/mercedesbenz/sechub/plugin/idea/sechubaccess/SecHubAccess.java

```diff
@@ -0,0 +1,71 @@
+package com.mercedesbenz.sechub.plugin.idea.sechubaccess;
+
+import com.mercedesbenz.sechub.api.DefaultSecHubClient;
+import com.mercedesbenz.sechub.api.SecHubClient;
+import com.mercedesbenz.sechub.api.SecHubClientException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.net.URI;
+
+public class SecHubAccess {
+
+    private static final Logger LOG = LoggerFactory.getLogger(SecHubAccess.class);
+    private static SecHubAccess INSTANCE;
+    private String secHubServerUrl;
+    private boolean trustAllCertificates;
+    private String userId;
+    private String apiToken;
```

Comment:
If it is required you could use a singleton here, but I think the following fields are not necessary here.
```suggestion

```
You could just update the `private SecHubClient client;` with a setter whenever the credentials or the server URL.change.